pkg/queue/request_metric.go

```diff
@@ -126,17 +130,21 @@ func (h *requestMetricsHandler) ServeHTTP(w http.ResponseWriter, r *http.Request
 			panic(err)
 		}
 		ctx := metrics.AugmentWithResponseAndRouteTag(h.statsCtx,
-			rr.ResponseCode, routeTag)
+			dw.ResponseCode, routeTag)
 		pkgmetrics.RecordBatch(ctx, requestCountM.M(1),
 			responseTimeInMsecM.M(float64(latency.Milliseconds())))
+		if h.reportLatencyInHeader {
+			dw.Header().Add("queue_proxy_request_latency_ms", strconv.Itoa(int(latency.Milliseconds())))
+		}
+		dw.RealFlush()
 	}()
 
-	h.next.ServeHTTP(rr, r)
+	h.next.ServeHTTP(dw, r)
```

Comment:
Normally `writer.WriteHeader` would be called inside this `ServeHTTP`, but `dw` makes sure `WriteHeader` is not called here. Because we want to call `WriteHeader` at line 139 above, after the header modification at line 137.